src/app/keyboard/components/virtual-keyboard-key/virtual-keyboard-key.component.ts

```diff
@@ -0,0 +1,44 @@
+import { ChangeDetectionStrategy, Component, HostBinding, Input } from '@angular/core'
+
+@Component({
+  selector: 'sofie-virtual-keyboard-key',
+  templateUrl: './virtual-keyboard-key.component.html',
+  styleUrls: ['./virtual-keyboard-key.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VirtualKeyboardKeyComponent {
+  @Input()
+  public key: string
+
+  @HostBinding('class')
+  public get keyClass(): string {
+    return this.key ? this.key.toLowerCase() : 'none'
+  }
+
+  @Input()
+  public label?: string
+
+  @Input()
+  @HostBinding('class.matched')
+  public isMatched: boolean = false
+
+  @Input()
+  public keyboardLayout: KeyboardLayoutMap
```

Comment:
Do a 'key' really need to know the entire layout of the keyboad?
You are only using it to get the 'mappedKey'. Maybe the 'mappedKey' should just be an input?

It makes more sense that a layout knows about its keys. Just like it's our Segments that knows about its Parts, not the Part that knows about its Segment.

Reply:
The variable name is not very descriptive. It is not the layout per se, but a map of key codes to key values.
I have moved the mapping logic out into the virtual keyboard, causing this variable to be removed.